core/src/main/java/org/apache/iceberg/avro/AvroSchemaVisitor.java

```diff
@@ -103,6 +109,10 @@ public T map(Schema map, T value) {
     return null;
   }
 
+  public T variant(Schema variant, List<T> fields) {
```

Comment:
Because we know that there are only two fields, I think it would be better to pass each individually as `T metadataResult` and `T valueResult`.